pennylane/templates/subroutines/grover.py

```diff
@@ -127,3 +129,21 @@ def expand(self):
                 Hadamard(wire)
 
         return tape
+
+    @classmethod
+    def _matrix(cls, *params):
+        num = cls.num_wires
```

Comment:
Hi @tgag17, slightly surprised that we'd get the correct number of wires here this way. :thinking: 

```python
import pennylane as qml

qml.templates.GroverOperator.num_wires
```
gives me
```
<WiresEnum.AnyWires: -1>
```

which is an internal placeholder we use to represent ops/templates that can take any number of wires. However, at this time, we'd already like to know the exact number of wires to use.

Maybe it's best to pass the number of wires to `_matrix` by [redefining the `matrix` method, see how it's done for QFT here](https://github.com/PennyLaneAI/pennylane/blob/742f2ff067d969a42e380968f36456c995dee2c9/pennylane/ops/qubit/non_parametric_ops.py#L1039).